File: backend/app/schemas/post.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


class CategorySimple(BaseModel):
    id: int
    name: str
    slug: str

    class Config:
        from_attributes = True


class PostBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    content: str = Field(..., min_length=1)
    author: str = Field(..., min_length=1, max_length=100)
    category_id: Optional[int] = None
    tags: Optional[str] = Field(None, max_length=200)
    language: Optional[str] = Field(None, max_length=20)
    is_markdown: bool = True


class PostCreate(PostBase):
    pass


class PostUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = Field(None, min_length=1)
    author: Optional[str] = Field(None, min_length=1, max_length=100)
    category_id: Optional[int] = None
    tags: Optional[str] = Field(None, max_length=200)
    language: Optional[str] = Field(None, max_length=20)
    is_markdown: Optional[bool] = None


class PostResponse(PostBase):
    id: int
    views: int
    category: Optional[CategorySimple] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class PostList(BaseModel):
    total: int
    page: int
    page_size: int
    posts: List[PostResponse]

File: frontend/types/post.ts
import { Category } from './category';

export interface Post {
  id: number;
  title: string;
  content: string;
  author: string;
  category_id?: number | null;
  category?: Category | null;
  tags?: string | null;
  language?: string | null;
  is_markdown: boolean;
  views: number;
  created_at: string;
  updated_at: string | null;
}

export interface PostCreate {
  title: string;
  content: string;
  author: string;
  category_id?: number;
  tags?: string;
  language?: string;
  is_markdown?: boolean;
}

export interface PostUpdate {
  title?: string;
  content?: string;
  author?: string;
  category_id?: number;
  tags?: string;
  language?: string;
  is_markdown?: boolean;
}

export interface PostList {
  total: number;
  page: number;
  page_size: number;
  posts: Post[];
}
